fdbserver/MoveKeys.actor.cpp

```diff
@@ -1424,6 +1435,11 @@ ACTOR static Future<Void> startMoveShards(Database occ,
 				state std::vector<Future<Void>> actors;
 
 				if (!currentKeys.empty()) {
+					// Pre validate consistency of update of keyServers and serverKeys
+					if (SERVER_KNOBS->AUDIT_DATAMOVE_PRE_CHECK) {
+						bool consistent = wait(auditKeyServersAndServerKeys(&tr, currentKeys));
```

Comment:
We only need to read `serverKeys` for the validation, since we are doing the `wait(krmGetRanges(&tr, keyServersPrefix` below anyway. Maybe validate the read result from `wait(krmGetRanges(&tr, keyServersPrefix`, and we will need to only read the corresponding `serverKeys` entries.

Reply:
Yes. This is done inside auditKeyServersAndServerKeys